lib/SIL/IR/Linker.cpp

```diff
@@ -418,6 +418,15 @@ void SILLinkerVisitor::visitMetatypeInst(MetatypeInst *MI) {
   linkInVTable(C);
 }
 
+void SILLinkerVisitor::visitGlobalAddrInst(GlobalAddrInst *GAI) {
+  if (!Mod.getASTContext().LangOpts.hasFeature(Feature::Embedded))
+    return;
+
+  SILGlobalVariable *G = GAI->getReferencedGlobal();
+  G->setDeclaration(false);
+  G->setLinkage(stripExternalFromLinkage(G->getLinkage()));
```

Comment:
Can you explain the motivation for this?

Reply:
Yes, the attached tests actually fail without this, because global variables from other modules do not actually get deserialized. I'll be happy to chat about options/alternatives, but IIUC currently deserialization of globals only produces declarations, see <https://github.com/apple/swift/blob/main/lib/SIL/IR/SILModule.cpp#L55>.

Reply:
Added the motivation into the PR description.